src/vstart.sh

```diff
@@ -1162,7 +1162,22 @@ EOF
 	fi
         ceph_adm config-key set mgr/cephadm/ssh_identity_key -i ~/.ssh/id_rsa
         ceph_adm config-key set mgr/cephadm/ssh_identity_pub -i ~/.ssh/id_rsa.pub
-        ceph_adm mgr module enable cephadm
+        local retry_flag=0
+        while [ $retry_flag -lt 7 ]
+        do
+            if [ `ceph_adm mgr stat | jq '.available'` = "true" ]; then
+                ceph_adm mgr module enable cephadm
+                break
+            else
+                retry_flag=$(($retry_flag + 1))
+                if [ $retry_flag -eq 7 ]; then
+                    debug echo "mgr is not available, exit."
+                    exit
+                fi
+                debug echo "mgr is not available, retry after 10 seconds ($retry_flag /6)"
+                sleep 10
+            fi
+        done
```

Comment:
I thik this code block probably would would be better to be on its own shell function, something like `wait_for_cephadm`

Reply:
Yes, thanks for the suggestion, I have added function.

Reply:
@Svelar thanks, I'm afraid that the name I proposed is not correct. Not sure why I thought that you were waiting for `cephadm` where in fact you are waiting for the mgr in this case. So maybe `waiting_for_mgr` would be better name for the function in this case. Sorry for that.

Reply:
No worries.